src/AggregateFunctions/AggregateFunctionEstimateCompressionRatio.cpp

```diff
@@ -54,7 +54,7 @@ struct AggregationFunctionEstimateCompressionRatioData
     UInt64 merged_compressed_size = 0;
     UInt64 merged_uncompressed_size = 0;
 
-    std::shared_ptr<NullWriteBuffer> null_buf;
+    std::unique_ptr<NullWriteBuffer> null_buf;
     std::shared_ptr<CompressedWriteBuffer> compressed_buf;
```

Comment:
Can this also be turned into a unique_ptr?
```suggestion
    std::unique_ptr<CompressedWriteBuffer> compressed_buf;
```

Reply:
> Would there be any concern about the buffer being accessed in parallel?

I am not sure if we access the buffer parallel or not, but I assume we don't. However, even if we would access the buffer parallel, `std::shared_ptr` doesn't help us with that. It is only about (shared) ownership and lifetime management, not about concurrent access.